<h1>Student Performance Indicator</h1>

### Life cycle of Machine learning Project

<ul>
    <li >Understand the problem statement
    <li>Data Collection</li>
    <li>Data Checks To Perform</li>
    <li>Exploratory Data Analysis</li>
    <li>Data Pre-Processing</li>
    <li>Model Training</li>
    <li>Choose the best model</li>
</ul>

<h3>1) Problem statement</h3>
<ul>
    <li> <h5>This project understands how the student's performance (test scores) is affected by other variables such as Gender, Ethnicity, Parental level of education, Lunch and Test preparation course.</h5></li>
</ul>

<h3>2) Data Collection</h3>
<ul>
    <li style='widith:5'>
        Dataset Source - <a href='https://www.kaggle.com/datasets/spscientist/students-performance-in-exams?datasetId=74977'>https://www.kaggle.com/datasets/spscientist
        students-performance-in-exams?datasetId=74977</a>
        The data consists of 8 column and 1000 rows.
    </li>
</ul>

<h5>2.1 Import Data and Required Packages</h5>

##### Importing Pandas, Numpy, Matplotlib, Seaborn and Warings Library.

In [19]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [20]:
a =pd.read_csv('stud.csv')

2.2) Dataset information
<ul>
<li>gender : sex of students -> (Male/female)</li>
<li>race/ethnicity : ethnicity of students -> (Group A, B,C, D,E)</li>
<li>parental level of education : parents' final education ->(bachelor's degree,some college,master's degree,associate's degree,high school)</li>
<li>lunch : having lunch before test (standard or free/reduced)</li>
<li>test preparation course : complete or not complete before test</li>
<li>math score</li>
<li>reading score</li>
<li>writing score</l>
</ul>

In [21]:
a.head()

,gender,race_ethnicity,parental_level_of_education,lunch,test_preparation_course,math_score,reading_score,writing_score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75


3) Data Checks to perform
<ul>
<li>Check Missing values</li>
<li>Check Duplicates</li>
<li>Check data type</li>
<li>Check the number of unique values of each column</li>
<li>Check statistics of data set</li>
<li>Check various categories present in the different categorical column</li>
</ul>

In [22]:
a.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race_ethnicity               1000 non-null   object
 2   parental_level_of_education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test_preparation_course      1000 non-null   object
 5   math_score                   1000 non-null   int64 
 6   reading_score                1000 non-null   int64 
 7   writing_score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [23]:
a.isna().sum()

gender                         0
race_ethnicity                 0
parental_level_of_education    0
lunch                          0
test_preparation_course        0
math_score                     0
reading_score                  0
writing_score                  0
dtype: int64

In [26]:
a.duplicated().sum()

np.int64(0)

In [25]:
a.nunique()

gender                          2
race_ethnicity                  5
parental_level_of_education     6
lunch                           2
test_preparation_course         2
math_score                     81
reading_score                  72
writing_score                  77
dtype: int64

In [27]:
a.describe()

,math_score,reading_score,writing_score
count,1000.00000,1000.000000,1000.000000
mean,66.08900,69.169000,68.054000
std,15.16308,14.600192,15.195657
min,0.00000,17.000000,10.000000
25%,57.00000,59.000000,57.750000
50%,66.00000,70.000000,69.000000
75%,77.00000,79.000000,79.000000
max,100.00000,100.000000,100.000000



<h4>Insight</h4>
<ul>
<li>From above description of numerical data, all means are very close to each other - between 66 and 68.05;</li>
<li>All standard deviations are also close - between 14.6 and 15.19;</li>
<li>While there is a minimum score 0 for math, for writing minimum is much higher = 10 and for reading myet higher = 17</li>
</ul>